File: dynamic_programming/1155_number_of_dice_rolls_with_target_sum/1155_number_of_dice_rolls_with_target_sum.py
class Solution:
    def numRollsToTarget(self, d: int, f: int, target: int) -> int:
        m = 10 ** 9 + 7
        dp = [[0] * (target + 1) for _ in range(d + 1)]
        dp[0][0] = 1
        for i in range(1, d + 1):
            for j in range(1, f + 1):
                for k in range(j, target + 1):
                    dp[i][k] = (dp[i][k] + dp[i - 1][k - j]) % m
        return dp[d][target]



d = 1
f = 6
target = 3
res = Solution().numRollsToTarget(d, f, target)
print(res)

File: dynamic_programming/1155_number_of_dice_rolls_with_target_sum/1155_number_of_dice_rolls_with_target_sum.go
package main 

import "fmt"


func numRollsToTarget(d int, f int, target int) int {
    mod := int((1e9)+7)
    dp := make([]int, target+1)
    dp[0] = 1
    for i := 0; i < d; i++ {
        ndp := make([]int, target+1)
        for j := 1; j <= f; j++ {
            for k := 0; k <= target-j; k++ {
                ndp[k+j] = (ndp[k+j]+dp[k])%mod 
            }
        }
        //fmt.Println(dp)
        dp = ndp
    }
    return dp[target]
}


func main() {
	d := 1
	f := 6
	target := 3
	res := numRollsToTarget(d, f, target)
	fmt.Println(res)
}